torax/sources/radiation_heat_sink.py

```diff
@@ -0,0 +1,124 @@
+"""Basic radiation heat sink for electron heat equation.."""
+
+import dataclasses
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from torax import array_typing
+from torax import geometry
+from torax import math_utils
+from torax import state
+from torax.config import runtime_params_slice
+from torax.sources import bremsstrahlung_heat_sink
+from torax.sources import qei_source
+from torax.sources import runtime_params as runtime_params_lib
+from torax.sources import source as source_lib
+from torax.sources import source_models as source_models_lib
+
+SOURCE_NAME = "radiation_heat_sink"
+
+
+def _radially_constant_fraction_of_Pin(
+    dynamic_runtime_params_slice: runtime_params_slice.DynamicRuntimeParamsSlice,
+    dynamic_source_runtime_params: runtime_params_lib.DynamicRuntimeParams,
+    geo: geometry.Geometry,
+    core_profiles: state.CoreProfiles,
+    source_models: source_models_lib.SourceModels,
+) -> jax.Array:
+    """Model function for radiation heat sink.
+
+    In this model, a fixed % of the total power input to the temp_el equation is lost."""
+    # Based on source_models.sum_sources_temp_el and source_models.calc_and_sum_sources_psi,
+    # but only summing over heating *input* sources (Pohm + Paux + Palpha)
+    # and summing over *both* ion and electron heating
+
+    def get_temp_el_profile(source_name: str, source: source_lib.Source) -> jax.Array:
+        # TODO: Currently this recomputes the profile for each source, which is inefficient
```

Comment:
agree. We will prioritize changing the pattern such that this won't happen.

Reply:
OK for the time being though?